client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -108,107 +75,65 @@ class FormBuilderModal extends Component {
    * @param {Function} submitFn The original submit function
    * @returns {Promise}
    */
-  handleSubmit(data, action, submitFn) {
-    this.clearResponse();
+  const handleSubmit = (data, action, submitFn) => {
+    clearResponse();
     let promise = null;
-    if (typeof this.props.onSubmit === 'function') {
-      promise = this.props.onSubmit(data, action, submitFn);
+    if (typeof onSubmit === 'function') {
+      promise = onSubmit(data, action, submitFn);
     } else {
       promise = submitFn();
     }
 
     if (promise) {
       // do not want this as part of the main promise chain.
       promise
-        .then((response) => {
-          if (response) {
-            this.setState({
-              response: response.message,
-              error: false,
-            });
+        .then((successResponse) => {
+          if (successResponse) {
+            setSuccess(successResponse.message);
           }
-          return response;
+          return successResponse;
         })
         .catch((errorPromise) => {
           errorPromise.then((errorText) => {
-            this.setState({
-              response: errorText,
-              error: true,
-            });
+            setFailure(errorText);
           });
         });
     } else {
       throw new Error('Promise was not returned for submitting');
     }
 
     return promise;
-  }
-
-  renderHeader() {
-    let { title } = this.props;
-    const { ModalHeaderComponent } = this.props;
-
-    if (title !== false) {
-      if (typeof title === 'object') {
-        // FormSchema title occasionaly contains html, only render text for modal title
-        const doc = new DOMParser().parseFromString(title.html, 'text/html');
-        title = doc.body.textContent || '';
+  };
+
+  const modalProps = pick(props, Object.keys(Modal.propTypes));
+  const formBuilderProps = pick(props, [
+    'schemaUrl', 'bodyClassName', 'autoFocus', 'onAction', 'identifier'
+  ]);
+
+  return (
+    <Modal {...modalProps} onClosed={handleHide}>
+      {response &&
+        <div className={error ? responseClassBad : responseClassGood}>
+          { castStringToElement('span', { html: response }) }
+        </div>
       }
-      return (
-        <ModalHeaderComponent toggle={this.handleHide}>{title}</ModalHeaderComponent>
-      );
-    }
-
-    if (this.props.showCloseButton === true && typeof this.props.onClosed === 'function') {
-      return (
-        <button
-          type="button"
-          className="close modal__close-button"
-          onClick={this.handleHide}
-          aria-label={i18n._t('Admin.CLOSE', 'Close')}
+      {schemaUrl &&
+        <FormBuilderLoaderComponent
+          {...formBuilderProps}
+          fieldHolder={{ className: classnames('modal-body', bodyClassName) }}
+          actionHolder={{ className: 'modal-footer' }}
+          onSubmit={handleSubmit}
+          onLoadingError={handleLoadingError}
```

Comment:
Ah right, different component.  Since we're defining props for modalProps and formBuilderProps should be consistent and also define formBuilderLoaderProps

```js
const formBuilderLoaderProps = {
  ...formBuilderProps,
  fieldHolder: { className: classnames('modal-body', bodyClassName) },
  actionHolder: { className: 'modal-footer' }
  onSubmit: handleSubmit
  onLoadingError: handleLoadingError
};

and then just go `<FormBuilderLoaderComponent {...formBuilderLoaderProps}>`